doc/smart-switch/high-availability/smart-switch-ha-hld.md

```diff
@@ -0,0 +1,2175 @@
+3. Ability to resume connections in the event of both planned and unplanned failover.
+4. After both planned and unplanned failover and recovery, the flow on all DPUs will be aligned.
+5. Only focus on the case, in which only 2 DPUs are in the HA set, although conceptually, we can have more than 2, but it is not in our current scope. 
+6. No complicated hashing logic in switches to ensure packets land on right switch.
+7. If switch receives a valid packet, it must not drop it due to flow replication delays.
+8. Ensure both inbound and outbound packets transit the same DPU for a given flow.
+
+### 3.2. Non-goals
+
+1. Handling data path failures perfectly is not one of the main goals of HA.
+    * Data plane failures can happen and affect our communications between NPU and DPUs, such as link flaps, traffic blackhole, but these issues will impact all traffic, not just smart switch scenarios. They are monitored in other ways and also should be addressed in a more generic way.
+    * Of course, when data path failure causes enough heartbeat or probes to be dropped, we will react on the change. But the goal is not providing a mechanism to detect and handle all gray failures due to ECMP setup. In short, if our probe says down, it is a down instead of 16% down.
+2. HA is not designed for providing perfect solutions to merge conflicting flows, but only ensures the flows will end up in a consistent state in all DPUs.
+    * This can happen in cases, such as switchover or when DPUs lose connection to each other and start to act as standalone. If policies in these DPUs don’t match, flow conflict can happen when later the connection recovers.
+    * To resolve the flow conflict, HA will do best effort, but it won’t be able to solve the problem alone. In cases like planned switchovers, our upstream service must coordinate and make sure policies are aligned.
+3. More than 2 DPUs participate in a single HA set.
+
+### 3.3. Assumptions
+
+When designing HA for SmartSwitch, we have a few assumptions:
+
+1. Stable decision on packet transformation: The same packet will always implement the same packet transformation as long as the DASH policies stay the same. Changes, such as random port selection, are not allowed in the current pipeline.
+2. No delay sync support: When creating a flow, we will replicate this flow to the standby DPU immediately with inline. The flow will not be truly created before receiving the ack from the standby DPU.
+3. No warm boot support: Warm boot is used to keep the data path continuing to work while upgrade happens. It is mostly used for upgrading switches that being considered as SPOF, as a gentler solution than a hard data plane down. However, it requires complicated coordination between many services on NPU and DPU. Gladly, since HA requires more than 1 switch to participate, we don’t need to worry about SPOF problem, hence we don’t need to consider warm boot support in our design.
```

Comment:
Suggest adding SPOF (Single Point of Failure) to terminology section

Reply:
thanks Chris! it is updated now.